src/sagemaker/session.py

```diff
@@ -3410,6 +3444,17 @@ def create_model(
             default_value=False,
             sagemaker_session=self,
         )
+        if container_defs:
+            update_list_of_dicts_with_values_from_config(
+                container_defs, MODEL_CONTAINERS_ENVIRONMENT_PATH, sagemaker_session=self
+            )
+        primary_container = resolve_nested_dict_value_from_config(
+            primary_container,
+            [ENVIRONMENT],
+            MODEL_PRIMARY_CONTAINER_ENVIRONMENT_PATH,
+            sagemaker_session=self
+        )
+
```

Comment:
Interesting, the note above seems to make this tricky here:
> If the list contains only one container, then
                it'll be passed to SageMaker Hosting as the ``PrimaryContainer`` and otherwise,
                it'll be passed as ``Containers``.

How should we figure out where to inject configs from for container_defs? Or check both locations? We might need to discuss this

Reply:
Also, if primary_container is deprecated, should we be injecting configs to it? I guess its nice to support it even so